### ISSUE TITLE
Project Reactor ServerRequestContext lost when using Mono.fromFeature()

### ISSUE THREAD

**Issue by denis111** [micronaut-projects/micronaut-core#5549]
When using project Reactor flow ServerRequestContext is lost when Mono is created by Mono.fromFuture. Here's the example code (i think it's too small to publish on github):

```java
@Controller("/context")
public class ContextController {

  @Get(value = "/flux", produces = MediaType.TEXT_PLAIN)
  Flux<String> flux() {
    return Mono
        .fromFuture(calculateAsync())
        .flux()
        .map(s -> s + " ")
        .flatMap(v -> Flux.just(v + (ServerRequestContext.currentRequest().isPresent() ? "ok" : "nok")));
  }

  @Get(value = "/flowable", produces = MediaType.TEXT_PLAIN)
  Flowable<String> flowable() {
    return Single
        .fromFuture(calculateAsync())
        .toFlowable()
        .map(s -> s + " ")
        .flatMap(v -> Flux.just(v + (ServerRequestContext.currentRequest().isPresent() ? "ok" : "nok")));
  }

  public CompletableFuture<String> calculateAsync() {
    CompletableFuture<String> completableFuture = new CompletableFuture<>();

    Executors.newCachedThreadPool().submit(() -> {
      Thread.sleep(500);
      completableFuture.complete("Hello ");
      return null;
    });

    return completableFuture;
  }
}
```

If you go to /context/flowable it's ok, context present but in /context/flux not.

### Task List

- [ x ] Steps to reproduce provided
- [ x ] Example that reproduces the problem uploaded to Github
- [ x ] Full description of the issue provided (see below)

### Steps to Reproduce

1. Go to /context/flux

### Expected Behaviour
The output should be Hello  ok

### Actual Behaviour
The output is Hello  nok

### Environment Information

- **Micronaut Version:** v2.5.5
- **JDK Version:** 11

**Comment by denis111** [comment]
If i change .fromFuture(calculateAsync()) to .fromCallable(() -> calculateAsync().get()) then it works fine

**Comment by graemerocher** [comment]
I think that is the way to go. Using `fromFuture` you would have to make sure the thread pool is instrumented

**Comment by denis111** [comment]
@graemerocher yes, I already changed everything in my "real" project to fromCallable. But with flowable it worked fine, the CompletableFuture there is not mine but returned by a third party library.

**Comment by graemerocher** [comment]
so in general as part of the shift to reactor we need to rewrite the instrumentation logic in reactor to use the context https://projectreactor.io/docs/core/release/api/reactor/util/context/Context.html

This work is not undertaken yet but will solve the issue you see for Reactor in the future

**Comment by denis111** [comment]
@graemerocher ok, thank you! I'll just use fromCallable for now.

**Comment by graemerocher** [comment]
@dstepanov this still an issue after your changes?

**Comment by MajaStojkovic-TomTom** [comment]
We notice a similar [issue](https://github.com/micronaut-projects/micronaut-azure/issues/319) with Mono response which we notice only occurs when the runtime is jetty or azure_function but for netty, it works fine. Are there any updates on the issue?

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by graemerocher** [comment]
Triage: classified as an HTTP server/Reactor context-propagation bug and left open pending a current verification. Core’s RouteExecutor installs ServerRequestContext through Reactor context for framework-managed publishers, but Mono.fromFuture() subscribes to a third-party CompletableFuture whose completion callback does not automatically carry Reactor context. Suggested fix: bridge the request context at the future boundary (for example with deferContextual/context capture or a supported context-propagation lifter), or document the required adapter for externally completed futures; add a regression test using the current Reactor integration and a non-framework executor.
